### ISSUE TITLE
Question on Multi-Class Segmentation on masks

### ISSUE THREAD

**Issue by sarmientoj24** [PyTorchLightning/lightning-bolts#796]
I saw that the `KittiDataset` in your U-Net semantic segmentation only has 1 channel for the mask input. But on your segmentation model, you have the output as `num_classes` which is 19. This means that it outpus 19 channels [here](https://github.com/PyTorchLightning/lightning-bolts/blob/22e494e546e7fe322d6b4cf36258819c7ba58e02/pl_bolts/models/vision/unet.py#L52)

Questions:
1. Is there a pre-processing/post processing that I was not able to see that transforms the **single-channel mask** input into an n_channel mask so that the cross-entropy makes sense? It seems that the following function only creates a single-channel mask where the background (and other unused classes) is 255.

```
    def encode_segmap(self, mask):
        """Sets void classes to zero so they won't be considered for training."""
        for voidc in self.void_labels:
            mask[mask == voidc] = self.ignore_index
        for validc in self.valid_labels:
            mask[mask == validc] = self.class_map[validc]
        # remove extra idxs from updated dataset
        mask[mask > 18] = self.ignore_index
        return 
```

I am referring to this training step where `mask` is only 1-channel while `out` is of `n_class` number of channels.
```
def training_step(self, batch, batch_nb):
        img, mask = batch
        img = img.float()
        mask = mask.long()
        out = self(img)
        loss_val = F.cross_entropy(out, mask, ignore_index=250)
        log_dict = {"train_loss": loss_val}
        return {"loss": loss_val, "log": log_dict, "progress_bar": log_dict}
```

2. Should the labels also include the background class? It seems like you did not include it in your examples.
3. How do I use DiceLoss?

**Comment by ethanwharris** [comment]
Hi @sarmientoj24, not sure about the other questions, but for number 1 the cross_entropy expects a long tensor as a target where the value in the ij'th position is the class at that position. E.g. for normal classification you have
```
preds: N x C, targets: N
```
and in 2D (segmentation) you have:
```
preds: N x C x W x H, targets: N x W x H
```
You can read more about the expected format here: https://pytorch.org/docs/1.9.1/generated/torch.nn.CrossEntropyLoss.html

Hope that helps :smiley:

**Comment by sarmientoj24** [comment]
That absolutely helps @ethanwharris 

Do you think that the background_class should be included?
